components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerV2.java

```diff
@@ -608,15 +608,17 @@ private String getSearchFilter(String attributeName, String filterOperation, Str
             throws BadRequestException {
 
         String searchFilter;
+        String formattedAttributeValue = formatSearchAttributeValue(filterOperation, attributeValue);
+        
         switch (attributeName) {
             case SCIMConstants.RoleSchemaConstants.DISPLAY_NAME_URI:
-                searchFilter = ROLE_NAME_FILTER_ATTRIBUTE + " " + filterOperation + " " + attributeValue;
+                searchFilter = ROLE_NAME_FILTER_ATTRIBUTE + " " + filterOperation + " " + formattedAttributeValue;
```

Comment:
Can you point out where these values are handled later in the flow?
Previously, the attribute value didn't include any operator-specific logic, but now with filteredAttribute, it does. I’d like to understand how this affects subsequent processing. Are there any changes in how these formatted values are utilized in query construction or filtering logic?

Reply:
Refer https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/606#issuecomment-2638754266.